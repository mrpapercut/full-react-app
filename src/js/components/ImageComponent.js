import {Component} from 'react';

import createElement from '~/util/createElement';

const [div, h2, span, img, em] = ['div', 'h2', 'span', 'img', 'em'].map(createElement);

class ImageComponent extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return div({
            className: 'imagecomponent'
        },
            h2({
                className: 'pagetitle'
            }, 'Image Component'),
            div({
                className: 'img-tag-image'
            },
                span({}, 'Image in <img>-tag'),
                img({
                    src: '/assets/images/incognitocat.png'
                }),
                em({}, 'I see you but you can\'t see me')
            ),
            div({
                className: 'background-image'
            },
                em({}, 'Image in CSS background')
            )
        );
    }
}

export default ImageComponent;
