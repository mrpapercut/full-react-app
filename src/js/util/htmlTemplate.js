const htmlTemplate = ({title, body, scriptTags, preloadedState}) => `<!doctype html>
<html>
<head>
    <meta charset="utf8"></meta>
    <title>${title}</title>
    <base href="/">

    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body>
    <div id="appwrapper">${body}</div>
    <script>${preloadedState}</script>
    ${scriptTags}
</body>
</html>
`;

export default htmlTemplate;
