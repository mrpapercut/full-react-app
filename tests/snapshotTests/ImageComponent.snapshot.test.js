import {createElement} from 'react';
import renderer from 'react-test-renderer';

import {wrapElementWithRouter} from '../util/snapshot-test-util';

import ImageComponent from '~/components/ImageComponent';

test('ImageComponent', () => {
    const component = renderer.create(wrapElementWithRouter(createElement(ImageComponent)));
    expect(component.toJSON()).toMatchSnapshot();
});
