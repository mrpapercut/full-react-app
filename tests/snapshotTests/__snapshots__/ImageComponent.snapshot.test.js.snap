// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageComponent 1`] = `
<div
  className="imagecomponent"
>
  <h2
    className="pagetitle"
  >
    Image Component
  </h2>
  <div
    className="img-tag-image"
  >
    <span>
      Image in &lt;img&gt;-tag
    </span>
    <img
      src="/assets/images/incognitocat.png"
    />
    <em>
      I see you but you can't see me
    </em>
  </div>
  <div
    className="background-image"
  >
    <em>
      Image in CSS background
    </em>
  </div>
</div>
`;
